custom/icds_reports/static/js/directives/map-or-sector-view/map-or-sector-view.directive.js

```diff
@@ -116,10 +116,22 @@ function MapOrSectorController($scope, $compile, $location, storageService, loca
         $compile(popup[0])($scope);
     };
 
+    // watching mapData for any changes. As soon as there is change in data, height of chart is calculated and
+    // chart is refreshed.
+    $scope.$watch(function () {
+        return vm.data.mapData;
+    }, function () {
+        if (vm.data.mapData && vm.data.mapData.chart_data) {
+            var chartHeight = vm.data.mapData.chart_data[0].values.length * 70;
+            vm.chartOptions.chart.height = ((chartHeight > 280) ? chartHeight : 280);
```

Comment:
where did 280 come from?

Reply:
We are giving height to the chart based on number of bars, But each bar doesn't take height of 70 each themselves, this was done keeping in mind that the calculated height includes the labels  and padding of the charts etc. So when there are less number of bars, padding and labels take their usual height, and bars get thinner resulting in overlapping location names. So When there are less than 4 horizontal bars, I gave them the height of a chart which is atleast the height of 4 bars one.
(This isnt the correct thing to do. Open for any suggestions)

Reply:
I have given the padding a certain value, which will always be added, irrespective of number of entries in the chart. I hope this is a bit better than earlier.

Reply:
sounds good, thanks